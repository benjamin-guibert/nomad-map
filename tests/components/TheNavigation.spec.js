import { createLocalVue, mount } from '@vue/test-utils'
import Vue from 'vue'
import ScrollTo from 'vue-scrollto'
import BootstrapVue from 'bootstrap-vue'
import TheNavigation from '@/components/TheNavigation.vue'

Vue.use(ScrollTo)

const localVue = createLocalVue()
localVue.use(BootstrapVue)

describe('TheNavigation', () => {
  const wrapper = mount(TheNavigation, {
    localVue,
    propsData: {
      travelLinks: {
        home: '#timeline-item-0',
        visited: '#timeline-item-1',
        current: '#timeline-item-2',
        next: '#timeline-item-3'
      }
    },
    stubs: {
      FontAwesomeIcon: true
    }
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
