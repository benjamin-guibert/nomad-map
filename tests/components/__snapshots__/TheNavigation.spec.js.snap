// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheNavigation renders 1`] = `"<nav class=\\"navigation my-colors-light fixed-top shadow\\"><ul class=\\"container justify-content-center nav\\"><li title=\\"Map\\" class=\\"nav-item navigation-link\\"><a target=\\"_self\\" href=\\"#world-map\\" class=\\"nav-link\\"><fontawesomeicon-stub icon=\\"map\\"></fontawesomeicon-stub> <p class=\\"navigation-link-label d-inline ml-2\\">Map</p></a></li> <li title=\\"Next\\" class=\\"nav-item navigation-link\\"><a target=\\"_self\\" href=\\"#timeline-item-3\\" class=\\"nav-link\\"><fontawesomeicon-stub icon=\\"bookmark\\"></fontawesomeicon-stub> <p class=\\"navigation-link-label d-inline ml-2\\">Next</p></a></li> <li title=\\"Current\\" class=\\"nav-item navigation-link\\"><a target=\\"_self\\" href=\\"#timeline-item-2\\" class=\\"nav-link\\"><fontawesomeicon-stub icon=\\"map-marker\\"></fontawesomeicon-stub> <p class=\\"navigation-link-label d-inline ml-2\\">Current</p></a></li> <li title=\\"Visited\\" class=\\"nav-item navigation-link\\"><a target=\\"_self\\" href=\\"#timeline-item-1\\" class=\\"nav-link\\"><fontawesomeicon-stub icon=\\"check-circle\\"></fontawesomeicon-stub> <p class=\\"navigation-link-label d-inline ml-2\\">Visited</p></a></li> <li title=\\"Home\\" class=\\"nav-item navigation-link\\"><a target=\\"_self\\" href=\\"#timeline-item-0\\" class=\\"nav-link\\"><fontawesomeicon-stub icon=\\"home\\"></fontawesomeicon-stub> <p class=\\"navigation-link-label d-inline ml-2\\">Home</p></a></li></ul></nav>"`;
